src/adapter/src/config/frontend.rs

```diff
@@ -162,28 +88,81 @@ impl SystemParameterFrontend {
     }
 }
 
-#[derive(Debug, Clone)]
-struct Metrics {
-    pub last_cse_time_seconds: UIntGauge,
-    pub last_sse_time_seconds: UIntGauge,
-    pub params_changed: IntCounter,
+impl Drop for SystemParameterFrontend {
+    fn drop(&mut self) {
+        tracing::info!("closing LaunchDarkly client");
+        self.ld_client.close();
```

Comment:
this method can potentially block for a while because it waits for pending analytics events to be sent ([code](https://docs.rs/launchdarkly-server-sdk/1.1.2/src/launchdarkly_server_sdk/client.rs.html#289-291)). Can we add another trace _after_ the call to close to make it easier to debug hangs?